proto/server/PI/proto/pi_server.h

```diff
@@ -22,15 +22,18 @@
 
 #include "stdint.h"
 
+#include "gnmi/gnmi.grpc.pb.h"
```

Comment:
Do you know if forward declaration syntax is supported for nested classes outside of class gNMI?
I'm getting the gcc error with "class gNMI::Service"

./PI/proto/pi_server.h:27:7: error: forward declaration of class cannot have a nested name specifier

See  http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2016/p0289r0.pdf

Reply:
Actually this header file is meant to be used in C code as well, so the forward declaration wouldn't work anyway (and even if it did it would probably not link without gNMI?)
You will have to make it a `void *`and include in the comment that it needs to be of type `gnmi::gNMI::Service` or `NULL` if you don't wish to provide a custom gNMI implementation. You will need to use `static_cast` in the implementation. You also cannot use optional parameters since this is a C header so in order to not break existing code, you will actually need to define a new function altogether, with a different name.

Reply:
Done